### PR TITLE
`General`: Anonymize collected events on user delete

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

We are already able to soft-delete users. Collected events are however neglected by the anonymization process.

### Description

This PR ensures, that the original login of the deleted user's identity of the event is anonymized.

### Steps for Testing

Prerequisites:
- 1 Admin
- 1 Students
- Artemis instance with Science-FeatureToggle enabled
- 1 Course with any Exercise

1. Log in to Artemis as a Student
2. Navigate to the Exercise of a Course
3. Open the Exercise (repeat 2/3 times, the DB now collected science events for the user)
4. Log in to Artemis as an Admin
5. Navigate to User Management and delete the Student
6. Look in the DB table `science_event` and make sure it doesn't contain a row with the student-login

#### Performance Review
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by rstief — DISMISSED** [review]
Code LGTM 👍

**Review by JohannesWt — APPROVED** [review]
Code 👍

File: src/main/java/de/tum/in/www1/artemis/service/user/UserService.java [hunk 1]
```diff
@@ -467,10 +473,11 @@ protected void anonymizeUser(User user) {
         final Set<String> originalGroups = user.getGroups();
         final String randomPassword = RandomUtil.generatePassword();
         final String userImageString = user.getImageUrl();
+        final String anonymizedLogin = lowerCase(RandomUtil.generateRandomAlphanumericString(), Locale.ENGLISH);
```
Comment (SimonEntholzer) [inline src/main/java/de/tum/in/www1/artemis/service/user/UserService.java:476]: do we really need to import this lowerCase function?
I guess the `"......".toLowerCase()` java standard library function would suffice here

**Review by pzdr7 — APPROVED** [review]
Tested locally.

After deleting, the events associated with the deleted user were anonymized ✅ 

image

**Review by rstief — APPROVED** [review]
Code LGTM. Tested locally, worked as expected.

**Review by JohannesStoehr — APPROVED** [review]
Code looks good to me. Also tested locally and the user was anonymised

**Review by pzdr7 — APPROVED** [review]
Reapprove after style fixes

**Review by bensofficial — APPROVED** [review]
Code changes look good to me 🚀

**Review by bassner — APPROVED** [review]

**Pull request merged**